File: apps/dsa-web/src/api/userFeedback.ts
import apiClient from './index';

export type UserFeedbackItem = {
  id: number;
  message: string;
  contact: string | null;
  portal_user_id: number | null;
  /** 门户登录用户提交时的注册邮箱 */
  portal_email: string | null;
  page_url: string | null;
  user_agent: string | null;
  created_at: string;
};

export type UserFeedbackListResponse = {
  total: number;
  page: number;
  limit: number;
  items: UserFeedbackItem[];
};

export async function fetchUserFeedbackList(params: {
  page?: number;
  limit?: number;
}): Promise<UserFeedbackListResponse> {
  const res = await apiClient.get<UserFeedbackListResponse>('/api/v1/system/user-feedback', {
    params: {
      page: params.page ?? 1,
      limit: params.limit ?? 50,
    },
  });
  return res.data;
}

File: api/v1/schemas/user_feedback.py
# -*- coding: utf-8 -*-
"""用户反馈 API 模型。"""

from typing import List, Optional

from pydantic import BaseModel, Field


class UserFeedbackSubmitRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=8000, description="反馈正文")
    contact: Optional[str] = Field(None, max_length=256, description="选填：联系方式或邮箱")
    page_url: Optional[str] = Field(
        None,
        max_length=512,
        description="选填：提交时页面 URL（前端可自动带上）",
    )


class UserFeedbackItem(BaseModel):
    id: int
    message: str
    contact: Optional[str] = None
    portal_user_id: Optional[int] = None
    portal_email: Optional[str] = Field(None, description="门户账号邮箱（提交时已登录门户时有值）")
    page_url: Optional[str] = None
    user_agent: Optional[str] = None
    created_at: str


class UserFeedbackListResponse(BaseModel):
    total: int
    page: int
    limit: int
    items: List[UserFeedbackItem]


class UserFeedbackSubmitResponse(BaseModel):
    id: int
    ok: bool = True
